File: scratch/replace_colors.js
const fs = require('fs');
const path = require('path');

const replacements = {
    "DarkBackground": "MaterialTheme.colorScheme.background",
    "DarkSurfaceVariant": "MaterialTheme.colorScheme.surfaceVariant",
    "DarkSurface": "MaterialTheme.colorScheme.surface",
    "TextPrimary": "MaterialTheme.colorScheme.onBackground",
    "TextSecondary": "MaterialTheme.colorScheme.onSurfaceVariant",
    "TextTertiary": "MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.5f)"
};

function processDirectory(directory) {
    const files = fs.readdirSync(directory);
    for (const file of files) {
        const fullPath = path.join(directory, file);
        if (fs.statSync(fullPath).isDirectory()) {
            processDirectory(fullPath);
        } else if (fullPath.endsWith('.kt') && !file.includes('Theme.kt') && !file.includes('Color.kt')) {
            let content = fs.readFileSync(fullPath, 'utf-8');
            let newContent = content;
            for (const [old, new_val] of Object.entries(replacements)) {
                newContent = newContent.split(old).join(new_val);
            }
            if (content !== newContent) {
                fs.writeFileSync(fullPath, newContent);
                console.log("Updated", fullPath);
            }
        }
    }
}

processDirectory('c:/Code/StudyTracker/app/src/main/java/com/iu/studytracker/ui');

File: scratch/replace_colors.py
import os
import glob

replacements = {
    "DarkBackground": "MaterialTheme.colorScheme.background",
    "DarkSurface": "MaterialTheme.colorScheme.surface",
    "DarkSurfaceVariant": "MaterialTheme.colorScheme.surfaceVariant",
    "TextPrimary": "MaterialTheme.colorScheme.onBackground",
    "TextSecondary": "MaterialTheme.colorScheme.onSurfaceVariant",
    "TextTertiary": "MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.5f)"
}

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    new_content = content
    for old, new in replacements.items():
        # Only replace if old is not in Theme.kt definition (we'll skip Theme.kt)
        new_content = new_content.replace(old, new)
        
    if new_content != content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Updated {filepath}")

for root, dirs, files in os.walk('c:\\Code\\StudyTracker\\app\\src\\main\\java\\com\\iu\\studytracker\\ui'):
    for file in files:
        if file.endswith('.kt') and file != 'Theme.kt' and file != 'Color.kt':
            process_file(os.path.join(root, file))
